### DISCUSSION TITLE
Startup Template for Module Available

### DISCUSSION THREAD

**Discussion by 274188A** [oqtane/oqtane.framework#682] [General]
Hi,
is there a visual studio or `dotnet` template available for starting a new module? thanks, 
JB

**Comment by mikecasas** [comment]
You can use oqtane itself to create a module, but I don't think a VS template exists. It shouldn't be hard to convert it though.  
The structure is at https://github.com/oqtane/oqtane.framework/tree/master/Oqtane.Server/wwwroot/Modules/Templates/External

**Comment by PizzaConsole** [comment]
I am actually currently working on this, and will have it ready in a few days.

**Comment by sbwalker** [comment]
There is a video series which explains how to create a module using the built in Module Creator which scaffolds a full set of code for you including build/packaging scripts:

https://www.youtube.com/playlist?list=PLYhXmd7yV0elLNLfQwZBUlM7ZSMYPTZ_f

And there are the sample modules:

https://github.com/oqtane/oqtane.blogs
https://github.com/oqtane/DNF.Projects

**Comment by PizzaConsole** [comment]
Still working on this, I just hit some delays.
